libpd_wrapper/z_libpd.c

```diff
@@ -277,11 +321,17 @@ int libpd_finish_message(const char *recv, const char *msg) {
 }
 
 void *libpd_bind(const char *sym) {
-  return libpdreceive_new(gensym(sym));
+  void *retval;
+  sys_lock();
+  retval = libpdreceive_new(gensym(sym));
+  sys_unlock();
```

Comment:
libpdreceive_new already locks internally, so this deadlocks as the first call to sys_lock grabs the mutex and the second lock cannot. Maybe the gensym call can be wrapped outside of the function call? This also makes we wonder if we should have a wrapped version of gensym, either as a macro or first level function.